include/RED4ext/Containers/StaticArray.hpp

```diff
@@ -0,0 +1,275 @@
+    {
+        Assign(aList);
+    }
+
+    template<std::input_iterator InputIt>
+    StaticArray(InputIt aFirst, InputIt aLast)
+    {
+        if (std::distance(aFirst, aLast) > MaxSize())
+            throw std::invalid_argument("StaticArray::StaticArray: Iterator range cannot exceed MaxSize");
+
+        Assign(aFirst, aLast);
+    }
+
+    ~StaticArray()
+    {
+        Clear();
+    }
+
+    Reference operator[](SizeType aIndex)
+    {
+        assert(aIndex < Size());
+        return entries[aIndex];
+    }
+
+    ConstReference operator[](SizeType aIndex) const
+    {
+        assert(aIndex < Size());
+        return entries[aIndex];
+    }
+
+    template<std::input_iterator InputIt>
+    void Assign(InputIt aFirst, InputIt aLast)
+    {
+        if (aFirst == aLast)
+        {
+            Clear();
+            return;
+        }
+
+        auto newSize = std::distance(aFirst, aLast);
+        if (newSize > MaxSize())
+            throw std::length_error("StaticArray::Assign: Iterator range cannot exceed MaxSize");
+
+        Resize(newSize);
+        std::copy(aFirst, aLast, entries);
+    }
+
+    void Assign(std::initializer_list<ValueType> aList)
+    {
+        if (aList.size() > MaxSize())
+            throw std::invalid_argument("StaticArray: Initializer list size cannot exceed MaxSize");
+
+        Assign(aList.begin(), aList.end());
+    }
+
+    void Assign(SizeType aSize, ConstReference aValue)
+    {
+        Resize(aSize);
+        std::fill(Begin(), End(), aValue);
+    }
+
+    [[nodiscard]] constexpr Reference At(SizeType aIndex)
+    {
+        if (aIndex >= Size())
+            throw std::out_of_range("StaticArray::At: Position out of range");
+
+        return entries[aIndex];
+    }
+
+    [[nodiscard]] constexpr ConstReference At(SizeType aIndex) const
+    {
+        if (aIndex >= Size())
+            throw std::out_of_range("StaticArray::At: Position out of range");
+
+        return entries[aIndex];
+    }
+
+    [[nodiscard]] Iterator Find(ConstReference aValue) noexcept
+    {
+        return Iterator(std::find(Begin(), End(), aValue));
+    }
+
+    [[nodiscard]] ConstIterator Find(ConstReference aValue) const noexcept
+    {
+        return ConstIterator(std::find(Begin(), End(), aValue));
+    }
+
+    [[nodiscard]] bool Contains(ConstReference aValue) const noexcept
+    {
+        return Find(aValue) != End();
+    }
+
+    [[nodiscard]] bool Contains(ConstIterator aPos) const noexcept
+    {
+        return Begin() <= aPos && aPos <= End();
+    }
+
+    [[nodiscard]] bool Contains(ConstIterator aFirst, ConstIterator aLast) const noexcept
+    {
+        return Begin() <= (std::min)(aFirst, aLast) && (std::max)(aLast, aFirst) <= End();
+    }
+
+    void Resize(SizeType aNewSize)
+    {
+        if (aNewSize > MaxSize())
+            throw std::invalid_argument("StaticArray::Resize: New size cannot exceed MaxSize");
+
+        if (aNewSize == size)
+            return;
+
+        size = aNewSize;
+    }
```

Comment:
It doesn't destroy elements. For example, if you have static array of handles (shared ptrs) and you clear the array, the handles will not be destroyed and will keep the object alive while being inaccessible.